File: clients/go/examples/test_client.go
package main

import (
	"fmt"
	"log"
	
	"github.com/blazecache/client"
)

func main() {
	fmt.Println("Testing BlazeCache Go client...")
	
	// Create client
	client, err := blazecache.New("127.0.0.1:6784")
	if err != nil {
		log.Fatal(err)
	}
	
	// Test ping
	if err := client.Ping(); err != nil {
		fmt.Printf("✗ Ping failed: %v\n", err)
	} else {
		fmt.Println("✓ Ping successful")
	}
	
	// Test set
	item := &blazecache.Item{
		Key:   "go-key",
		Value: []byte("go-value"),
	}
	
	if err := client.Set(item); err != nil {
		fmt.Printf("✗ Set failed: %v\n", err)
	} else {
		fmt.Println("✓ Set successful")
	}
	
	// Test get
	result, err := client.Get("go-key")
	if err != nil {
		if err == blazecache.ErrNotFound {
			fmt.Println("✗ Key not found")
		} else {
			fmt.Printf("✗ Get failed: %v\n", err)
		}
	} else {
		fmt.Printf("✓ Get successful: %s\n", string(result.Value))
	}
	
	// Test get non-existent key
	_, err = client.Get("nonexistent")
	if err == blazecache.ErrNotFound {
		fmt.Println("✓ Correctly returned not found for missing key")
	} else if err != nil {
		fmt.Printf("✗ Get failed: %v\n", err)
	} else {
		fmt.Println("✗ Should not have found key")
	}
	
	// Test delete
	if err := client.Delete("go-key"); err != nil {
		if err == blazecache.ErrNotFound {
			fmt.Println("✗ Key not found for delete")
		} else {
			fmt.Printf("✗ Delete failed: %v\n", err)
		}
	} else {
		fmt.Println("✓ Delete successful")
	}
	
	// Test multi-get
	client.Set(&blazecache.Item{Key: "key1", Value: []byte("value1")})
	client.Set(&blazecache.Item{Key: "key2", Value: []byte("value2")})
	
	results, err := client.GetMulti([]string{"key1", "key2", "key3"})
	if err != nil {
		fmt.Printf("✗ Multi-get failed: %v\n", err)
	} else {
		fmt.Printf("✓ Multi-get successful: %d keys found\n", len(results))
		for key, item := range results {
			fmt.Printf("  %s: %s\n", key, string(item.Value))
		}
	}
	
	fmt.Println("\n✅ Go client test completed!")
}

File: clients/rust/examples/test_client.rs
use blazecache_client::{TcpClient, SelectionStrategy};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("Testing BlazeCache Rust client...");
    
    // Create client with round robin strategy
    let servers = vec!["127.0.0.1:8080".to_string()];
    let client = TcpClient::new(servers);
    
    // Test ping
    match client.ping().await {
        Ok(_) => println!("✓ Ping successful"),
        Err(e) => println!("✗ Ping failed: {}", e),
    }
    
    // Test set
    match client.set("rust-key", b"rust-value".to_vec()).await {
        Ok(_) => println!("✓ Set successful"),
        Err(e) => println!("✗ Set failed: {}", e),
    }
    
    // Test get
    match client.get("rust-key").await {
        Ok(Some(value)) => println!("✓ Get successful: {}", String::from_utf8_lossy(&value)),
        Ok(None) => println!("✗ Key not found"),
        Err(e) => println!("✗ Get failed: {}", e),
    }
    
    // Test get non-existent key
    match client.get("nonexistent").await {
        Ok(Some(_)) => println!("✗ Should not have found key"),
        Ok(None) => println!("✓ Correctly returned None for missing key"),
        Err(e) => println!("✗ Get failed: {}", e),
    }
    
    // Test delete
    match client.delete("rust-key").await {
        Ok(true) => println!("✓ Delete successful"),
        Ok(false) => println!("✗ Key not found for delete"),
        Err(e) => println!("✗ Delete failed: {}", e),
    }
    
    // Test multi-get
    let _ = client.set("key1", b"value1".to_vec()).await;
    let _ = client.set("key2", b"value2".to_vec()).await;
    
    match client.get_multi(&["key1", "key2", "key3"]).await {
        Ok(results) => {
            println!("✓ Multi-get successful: {} keys found", results.len());
            for (key, value) in results {
                println!("  {}: {}", key, String::from_utf8_lossy(&value));
            }
        },
        Err(e) => println!("✗ Multi-get failed: {}", e),
    }
    
    println!("\n✅ Rust client test completed!");
    
    Ok(())
}
